packages/concerto-core/test/serializer/maptype/serializer.js

```diff
@@ -0,0 +1,926 @@
+            let concepts = factory.newConcept('org.acme.sample', 'Concepts');
+
+            const bob = factory.newConcept('org.acme.sample', 'Person', 'Bob');
+            const alice = factory.newConcept('org.acme.sample', 'Person', 'Alice');
+
+            concepts.reservation = new Map();
+            concepts.reservation.set('$class', 'org.acme.sample.Reservation');
+            concepts.reservation.set(bob, '2023-10-28T01:02:03Z');
+            concepts.reservation.set(alice, '2023-11-28T01:02:03Z');
+
+            const json = serializer.toJSON(concepts);
+
+            json.should.deep.equal({
+                $class: 'org.acme.sample.Concepts',
+                reservation: {
+                    $class: 'org.acme.sample.Reservation',
+                    '{"$class":"org.acme.sample.Person","name":"Bob"}' : '2023-10-28T01:02:03Z',
+                    '{"$class":"org.acme.sample.Person","name":"Alice"}' : '2023-11-28T01:02:03Z'
+                }
+            });
+        });
+
+        it('should generate a JSON object with a Map <Concept, Boolean>, where Concept is Identified', () => {
+            let concepts = factory.newConcept('org.acme.sample', 'Concepts');
+
+            const bob = factory.newConcept('org.acme.sample', 'Person', 'Bob');
+            const alice = factory.newConcept('org.acme.sample', 'Person', 'Alice');
+
+            concepts.vip = new Map();
+            concepts.vip.set('$class', 'org.acme.sample.GuestList');
+            concepts.vip.set(bob, true);
+            concepts.vip.set(alice, true);
+
+            const json = serializer.toJSON(concepts);
+
+            json.should.deep.equal({
+                $class: 'org.acme.sample.Concepts',
+                vip: {
+                    $class: 'org.acme.sample.GuestList',
+                    '{"$class":"org.acme.sample.Person","name":"Bob"}' : true,
+                    '{"$class":"org.acme.sample.Person","name":"Alice"}' : true
+                }
+            });
+        });
+
+        it('should generate a JSON object with a Map <Concept, Concept>, where Concept is Identified', () => {
+            let concepts = factory.newConcept('org.acme.sample', 'Concepts');
+
+            const bob = factory.newConcept('org.acme.sample', 'Person', 'Bob');
+            const alice = factory.newConcept('org.acme.sample', 'Person', 'Alice');
+
+            concepts.marriages = new Map();
+            concepts.marriages.set('$class', 'org.acme.sample.MarriageRegister');
+            concepts.marriages.set(bob, alice);
+
+            const json = serializer.toJSON(concepts);
+
+            json.should.deep.equal({
+                $class: 'org.acme.sample.Concepts',
+                marriages: {
+                    $class: 'org.acme.sample.MarriageRegister',
+                    '{"$class":"org.acme.sample.Person","name":"Bob"}' : '{"$class":"org.acme.sample.Person","name":"Alice"}'
```

Comment:
This doesn't feel write to me. Certainly for the value. 

Should the key be the identifier for the Concept instead?

Reply:
As discussed, dropping support for this until we agree on a concrete design. @dselman FYI